boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -528,12 +702,31 @@ WinLdrSetProcessorContext(void)
     /* Re-initialize EFLAGS */
     __writeeflags(0);
 
-    /* Set the PDBR */
-    __writecr3((ULONG_PTR)PDE);
+    if (!PaeModeOn)
+    {
+        /* Set the PDBR */
+        __writecr3((ULONG_PTR)PDE);
+    }
+    else
+    {
+        TRACE("WinLdrSetProcessorContext: PaeTables %X, cr0 %X, cr3 %X, cr4 %X\n", PaeTables, __readcr0(), __readcr3(), __readcr4());
+
+        /* Flush TLB */
+        __writecr3(__readcr3());
+
+        /* Set the PDPTR */
+        __writecr3((ULONG_PTR)PaeTables);
```

Comment:
This flushes the TLB as well, the line above should not be needed

Reply:
Ok